### ISSUE TITLE
РГР Этап 2

### ISSUE THREAD

**Issue by sokolnikovalexey** [EldarMuhamethanov/OP#7]
Программа падает, при разных входных данных. Чтобы проще тестировать - установил длину дерева 2 и вот что происходит
Для данных:
![image](https://user-images.githubusercontent.com/1480990/86789349-dffca200-c06f-11ea-81d6-b06ca2a8cbab.png)
Все хорошо.

Для данных
![image](https://user-images.githubusercontent.com/1480990/86789423-f4d93580-c06f-11ea-99cf-3b544067c0a8.png)
Падает ошибка при выполнении
![image](https://user-images.githubusercontent.com/1480990/86789477-028ebb00-c070-11ea-890f-1f72ba085c59.png)

**Comment by EldarMuhamethanov** [comment]
![image](https://user-images.githubusercontent.com/61377022/86791109-dd9b4780-c071-11ea-8631-025a4cac5616.png)

Если длина дерева, это количество узлов максимальное, то у меня все нормально работает при максимальном количестве 2

**Comment by sokolnikovalexey** [comment]
> ![image](https://user-images.githubusercontent.com/61377022/86791109-dd9b4780-c071-11ea-8631-025a4cac5616.png)
> 
> Если длина дерева, это количество узлов максимальное, то у меня все нормально работает при максимальном количестве 2

Вы хотите сказать, что проблема в моей операционке или компиляторе?

**Comment by EldarMuhamethanov** [comment]
> > ![image](https://user-images.githubusercontent.com/61377022/86791109-dd9b4780-c071-11ea-8631-025a4cac5616.png)
> > Если длина дерева, это количество узлов максимальное, то у меня все нормально работает при максимальном количестве 2
> 
> Вы хотите сказать, что проблема в моей операционке или компиляторе?


Нет, я лишь хочу сказать что у меня такая проблема почему то не появляется
Могу, если что сделать запись экрана

**Comment by sokolnikovalexey** [comment]
Поднял у себя windows и использовал компилятор ваш, проблема пропала, задание принято
